File: common/src/main/java/generations/gg/generations/core/generationscore/world/level/block/GenerationsWallHangingSignBlock.java
package generations.gg.generations.core.generationscore.world.level.block;

import generations.gg.generations.core.generationscore.world.level.block.entities.PokeModHangingSignBlockEntity;
import net.minecraft.core.BlockPos;
import net.minecraft.world.level.block.WallHangingSignBlock;
import net.minecraft.world.level.block.entity.BlockEntity;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.level.block.state.properties.WoodType;
import org.jetbrains.annotations.NotNull;

public class GenerationsWallHangingSignBlock extends WallHangingSignBlock {
	public GenerationsWallHangingSignBlock(Properties arg, WoodType arg2) {
		super(arg, arg2);
	}

	@Override
	public BlockEntity newBlockEntity(@NotNull BlockPos arg, @NotNull BlockState arg2) {
		return new PokeModHangingSignBlockEntity(arg, arg2);
	}
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/GenerationsWallHangingSignBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block

import generations.gg.generations.core.generationscore.common.world.level.block.entities.GenerationsHangingSignBlockEntity
import net.minecraft.core.BlockPos
import net.minecraft.world.level.block.WallHangingSignBlock
import net.minecraft.world.level.block.entity.BlockEntity
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.level.block.state.properties.WoodType

class GenerationsWallHangingSignBlock(arg: WoodType, arg2: Properties) :
    WallHangingSignBlock(arg, arg2) {
    override fun newBlockEntity(arg: BlockPos, arg2: BlockState): BlockEntity? {
        return GenerationsHangingSignBlockEntity(arg, arg2)
    }
}
